### PR TITLE
Fix cache result of external grammar info.

### PR DESCRIPTION
**Pull request by angelozerr**
Fix cache result of external grammar info.

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
This PR uses the (none used referencedExternalGrammarInitialized  flag) to avoid computing the information if DOM document has an external grammar or not.

 No need to write test, because we have test for that.

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMDocument.java [hunk 1]
```diff
@@ -349,32 +349,37 @@ private synchronized boolean intializeExternalGrammar() {
...
-			// None grammar found with standard mean, check if it some components like XML
-			// file associations bind this XML document to a grammar with external schema
-			// location.
-			try {
-				externalSchemaLocation = resolverExtensionManager.getExternalSchemaLocation(new URI(getDocumentURI()));
-				if (externalSchemaLocation != null) {
-					return true;
+		try {
+			if (resolverExtensionManager != null) {
+				// None grammar found with standard mean, check if it some components like XML
```
Comment (fbricon) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMDocument.java:354]: No grammar

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMDocument.java [hunk 2]
```diff
@@ -349,32 +349,37 @@ private synchronized boolean intializeExternalGrammar() {
...
-			// it some components like XML
-			// Catalog, XSL and XSD resolvers, etc bind this XML document to a grammar.
-			// Get root element
-			DOMElement documentElement = getDocumentElement();
-			if (documentElement == null) {
-				return false;
-			}
-			String namespaceURI = documentElement.getNamespaceURI();
-			return resolverExtensionManager.resolve(getDocumentURI(), namespaceURI, null) != null;
+				// None grammar found with standard mean and external schema location, check if
```
Comment (fbricon) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMDocument.java:367]: No grammar

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMDocument.java (hunk 1 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMDocument.java]: fixed

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMDocument.java (hunk 2 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMDocument.java]: fixed

**Pull request merged**